providers/default.rb

```diff
@@ -224,11 +231,41 @@ def move_opt_chef(src, dest)
   end
 end
 
+def prepare_windows
+  Kernel.spawn('c:/windows/system32/schtasks.exe /F /RU SYSTEM /create /sc minute /mo 1 /tn Chef_upgrade /tr "powershell.exe c:/opscode/chef_upgrade.ps1"') if platform_family?('windows')
+  FileUtils.rm_rf "#{chef_install_dir}/bin/chef-client.bat" if platform_family?('windows') && platform_family?('windows')
+end
+
 def execute_install_script(install_script)
   if windows?
     powershell_script 'name' do
       code <<-EOH
-    #{install_script}
+	  $command = {
+		Get-Service chef-client | stop-service
```

Comment:
This will write a non-terminating error if chef is not running as a service (for example, using the scheduled task mechanism).  This will cause the later execution to be unhappy.

Reply:
I changed these to ignore errors on Get-Service.